src/config/scrivitoSites.ts

```diff
@@ -84,24 +85,40 @@ export function isNoSitePresent(): boolean {
 }
 
 export async function ensureSiteIsPresent() {
-  if ((await load(currentSiteId)) === null) {
-    if (await load(isNoSitePresent)) {
-      ensureUserIsLoggedIn()
-      return
-    }
-
-    navigateTo(() => {
-      const websites = appWebsites() || []
-      const preferredLanguageOrder = [...window.navigator.languages, 'en', null]
-
-      for (const language of preferredLanguageOrder) {
-        const site = websites.find((site) => siteHasLanguage(site, language))
-        if (site) return site
-      }
-
-      return websites[0] || null
-    })
+  if (await load(currentSiteId)) return
+
+  if (await load(isNoSitePresent)) {
+    ensureUserIsLoggedIn()
+    return
+  }
+
+  const site = await load(getPreferredSite)
+  if (!site) return
+
+  const language = languageForUrl(window.location.href)
+  const slashesCount = language ? 2 : 1
+  const prefixLength = (language?.length || 0) + slashesCount
+  const path = window.location.pathname.substring(prefixLength)
```

Comment:
Why do we need this logic? It looks quite complex to me.

Reply:
If we don't need it the logic could swap `location.origin` from `location.href` with `urlFor(site)`.

Reply:
Depends on which kinds of old permalinks we want to handle:
1. links after relaunching a single-language site
2. links to the primary language site version that had no language prefix before the relaunch
3. links to a language version that no longer exists after the relaunch

The current logic handles all cases. If we drop (3) the logic is simpler.

Reply:
I would drop (3) to keep it more simple.